### PR TITLE
add rows properties

### PR DESCRIPTION
**Pull request by huachaohuang**
Collect rows count from CF_WRITE.

### CODE REVIEW

File: src/util/properties.rs [hunk 1]
```diff
@@ -179,9 +200,95 @@ pub struct IndexHandle {
...
+            try!(buf.read_exact(&mut k));
+            let mut v = IndexHandle::default();
+            v.size = try!(buf.decode_u64());
+            v.offset = try!(buf.decode_u64());
+            res.insert(k, v);
+        }
+        Ok(IndexHandles(res))
+    }
+
+    fn get_approximate_diff_in_range(&self, start: &[u8], end: &[u8]) -> u64 {
```
Comment (siddontang) [inline src/util/properties.rs]: row_distance may be better than diff?

File: src/util/properties.rs [hunk 2]
```diff
@@ -179,9 +200,95 @@ pub struct IndexHandle {
...
+        let mut range = self.range::<[u8], _>((Included(start), Unbounded));
+        let start_offset = match range.next() {
```
Comment (siddontang) [inline src/util/properties.rs]: seem it is not offset, it is row count?

Comment (huachaohuang) [inline src/util/properties.rs]: This is a general struct shared rows and size properties.

Comment (siddontang) [inline src/util/properties.rs]: got it, but I still think we should use a better name instead of diff.

Comment (huachaohuang) [inline src/util/properties.rs]: How about `get_approximate_distance_in_range`?

Comment (siddontang) [inline src/util/properties.rs]: +1

**Comment by siddontang** [comment]
LGTM

PTAL @BusyJay @zhangjinpeng1987

File: src/util/properties.rs [hunk 3]
```diff
@@ -90,38 +89,43 @@ impl MvccProperties {
...
     }
 }
 
 pub struct MvccPropertiesCollector {
     props: MvccProperties,
     last_row: Vec<u8>,
+    num_errors: u64,
     row_versions: u64,
+    index_handle: IndexHandle,
+    index_handles: IndexHandles,
```
Comment (zhangjinpeng1987) [inline src/util/properties.rs]: index_handle and index_handles a bit of confusion.

Comment (huachaohuang) [inline src/util/properties.rs]: How about `cur_index_handle` and `row_index_handles`?

**Comment by huachaohuang** [comment]
PTAL @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**